### PR TITLE
feat(IBMService): enhance vcap parsing

### PR DESCRIPTION
**Pull request by jorge-ibm**
ref: https://github.ibm.com/arf/planning-sdk-squad/issues/1096

Changes:
- enhanced vcap parsing to follow the latest service factory [design](https://github.ibm.com/CloudEngineering/openapi-sdkgen/wiki/Design-Of-Service-Config-Changes#external-config-behavior)
- Added unit tests for new vcap parsing logic
- Updated folder path for unit tests to support local test runs on mac (thanks @mediumTaj)

### CODE REVIEW

File: src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs [hunk 1]
```diff
@@ -83,27 +83,42 @@ private static VcapCredential GetVcapCredentialsObject(Dictionary<string, List<V
...
+            {
+                if (credentials == null || credentials.Count == 0)
+                {
+                    return null;
+                }
+                if (!string.IsNullOrEmpty(plan))
+                {
+                    foreach (VcapCredential credential in credentials)
+                    {
+                        if (credential.Plan.ToLower() == plan.ToLower())
```
Comment (jorge-ibm) [inline src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: @padamstx @mediumTaj - I was skeptical about leaving this here. I want to say we don't check the plan in the other languages, but I may be wrong. Is this still being used?

Comment (mediumTaj) [reply src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: heh I think I copied this from Java. Looks like it was removed.

https://github.com/IBM/java-sdk-core/commit/f40c95229d3c86e7f8f76ba16cf88fb146500936#diff-5500372b768c88ebe4772f4eb00299afL148

I was trying to stay as close to Java as possible so if we don't use it there I think it's OK to remove in .NET as well

Comment (jorge-ibm) [reply src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: Great, just wanted to double check with you. I'll go ahead and remove that check then

File: test/IBM.Cloud.SDK.Core.Tests/CredentialUtilsTests.cs [hunk 2]
```diff
@@ -155,7 +492,7 @@ public void TestCredentialOrder()
             string[] lines = { "TEST_SERVICE_LOCATION=user-set-location" };
 
             var directoryPath = Environment.GetFolderPath(
-                Environment.SpecialFolder.CommonApplicationData
+                Environment.SpecialFolder.LocalApplicationData
```
Comment (mediumTaj) [inline test/IBM.Cloud.SDK.Core.Tests/CredentialUtilsTests.cs:495]: 👍

**Review by mediumTaj — APPROVED** [review]
👍 looks good! tests are passing in the Watson SDK using this core.

**Review by padamstx — APPROVED** [review]
LGTM

**Review by mkistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**